include/swift/Remote/ExternalTypeRefCacheProvider.h

```diff
@@ -0,0 +1,74 @@
+//===--- ExternalTypeRefCacheProvider.h - Abstract access to external caches of
+//typeref ------*- C++ -*-===//
```

Comment:
The `ExternalTypeInfoProvider` is called so because it provides typeinfo records from the outside. Here, I think `ExternalTypeRefCache` alone would suffice (without the `provider)?

Reply:
Thinking about it ... do we even need the `External` in the type name? I.e.: we could pass around a `TypeRefCache *ExternalTypeRefCache`

Reply:
I agree with removing "provider", I do think "external" in the type name makes it more obvious what it is, versus as the variable name.